sentry-android-core/src/main/java/io/sentry/android/core/ActivityFramesTracker.java

```diff
@@ -57,7 +60,24 @@ public synchronized void addActivity(final @NotNull Activity activity) {
     if (!isFrameMetricsAggregatorAvailable()) {
       return;
     }
-    frameMetricsAggregator.add(activity);
+
+    try {
+      if (MainThreadChecker.isMainThread()) {
```

Comment:
`AppLifecycleIntegrationTest` has tests that it checks if its executed in the main thread or not.